### ISSUE TITLE
Crashes in latest stable, 2.1.177

### ISSUE THREAD

**Issue by malx122** [Starcounter/Starcounter#3046]
Having random server crashes in 2.1.177. I can not find any patterns, it does not happen often and there is no evident cause by me triggering it. (It always happened on a full reload of a random webpage). I do however get som things in the log:
```
Access violation (C0000005) detected at ..\..\src\sccoredb\sccoredb\sccoredb.cpp:1289 (Thread 12568).
    0000006CB52826DD sccoredb!update::Rregister_bad_ref_read : u_badrefs.cpp (152)
    0000006CB529F5AF sccoredb!evaluate_reference_value : objread.cpp (1508)
    0000006CB529FF68 sccoredb!_object_read_committed_objref : objread.cpp (921)
    0000006CB52A073E sccoredb!_object_read_objref : objread.cpp (403)
    0000006CB527E73B sccoredb!star_get_reference : sccoredb.cpp (1280)
    00007FFA965CF0E1 <unknown>!
    00000000000089AD <unknown>!
    00000004001F037E <unknown>!
    000000000000000C <unknown>!
    00000068094ADBF8 <unknown>!
    00000068094ADC00 <unknown>!
    00000068094ADBF0 <unknown>!
    00000A3F9EF7CC55 <unknown>!
    00007FFAF6123588 clr!
    00000068094AE058 <unknown>!
    00007FFA964A0310 <unknown>!
    000000696C2FCFD0 <unknown>!
    00000068094ADB80 <unknown>!
    00007FFA965CF0E1 <unknown>!
    00000068094ADC40 <unknown>!
    0000000000000030 <unknown>!
    0000000000007CCC <unknown>!
    000000040028D38D <unknown>!
```

**Comment by per-samuelsson** [comment]
@malx122 

Thanks for reporting this. Can't recall I've seen anything like it lately, so I can't provide much info straight away, but I'll surely make sure we take a look at it.

How often is "does not happen often" approx? Can you isolate it in some smaller context, or is this when you are running full application suite?

**Comment by malx122** [comment]
Never tried "full suite", normally run launcher, signin (our modified version) and the application I'm developing on for the moment. Has happened maybe once a day last 5-6 work days with different applications running. (When developing I mostly just restarting the app I'm working with but sometimes I restart all if I have to do something in our shared library that also SignIn uses)

**Comment by per-samuelsson** [comment]
OK, thanks @malx122. I'm a little unsure how we should approach this going further, since its somewhat nondeterministic. We'll back to you on that. Maybe @eriohl has some idea on this?

**Comment by eriohl** [comment]
Haven't seen this before. Thank you for reporting. Always the same stack-trace?

**Comment by malx122** [comment]
Tried to look back but could only see the latest in the log. But as I remember it is always the same stack-trace (otherwise very similair)

**Comment by eriohl** [comment]
OK. Thank you. If you could confirm this next time you see this it would be appriciated. I will check the code see if I can discover what might be wrong (no recent updates to this code).

**Comment by eriohl** [comment]
Found the problem (most likely). Fix will be included in next build (2.1.470). Please re-open the issue if you seen something similar after upgrade.
